r/R/dataset.R

```diff
@@ -438,3 +438,20 @@ stop_if_no_datasets <- function() {
     stop("This build of the arrow package does not support Datasets", call. = FALSE)
   }
 }
+
+#' Add filename as dataset column
+#'
```

Comment:
This probably should have more documentation, noting what it returns, what it does, how it (presumably) only does anything if you're doing a query on a Dataset, etc.